store/store.go

```diff
@@ -322,20 +345,41 @@ func (bs *BlockStore) PruneBlocks(height int64) (uint64, error) {
 				return 0, err
 			}
 		}
+
+		// TODO: batch these txs and flush them when we flush the blocks
+		block := bs.LoadBlock(h)
+		for _, tx := range block.Txs {
+			txKey := calcTxHashKey(tx.Hash())
+			txValue, err := bs.db.Get(txKey)
+			if err != nil {
+				return 0, err
+			}
+			if txValue != nil {
+				if err := batchTxs.Delete(txKey); err != nil {
+					return 0, err
+				}
+			}
+		}
```

Comment:
This will need to be done prior to deleting the block else there won't be any block to query